src/Pos/Explorer/Web/Server.hs

```diff
@@ -532,9 +554,23 @@ unwrapOrThrow = either (throwM . Internal) pure
 
 -- | Get transaction from memory (STM) or throw exception.
 fetchTxFromMempoolOrFail :: ExplorerMode m => TxId -> m TxAux
-fetchTxFromMempoolOrFail txId =
-    view (mpLocalTxs . at txId) <$> getMemPool >>=
-    maybeThrow (Internal "Transaction not found in the mempool")
+fetchTxFromMempoolOrFail txId = do
+    memPoolTxs        <- localMemPoolTxs
+    let memPoolTxsSize = HM.size memPoolTxs
+
+    logDebug $ sformat ("Mempool size "%int%" found!") memPoolTxsSize
```

Comment:
What I've seen that you prefer to define local variables very often. Sometimes it is helpful, sometimes not. 

For example: Does it make sense to create `let memPoolTxsSize` to store `HM.size memPoolTxs` just for using in `logDebug`? I think it's fine to define logDebug without this `memPoolTxsSize`: 
``` 
logDebug $ sformat ("Mempool size "%int%" found!") $ HM.size memPoolTxs
```
That's ^ readable and understandable. And one line less code. The same with `maybeTxAux` and on other places. 

Just my personal view - but it does not criticize the code, which looks pretty nice in general!